### ISSUE TITLE
emits 'row' after 'end' sometimes

### ISSUE THREAD

**Issue by tomfa** [ffalt/xlsx-extract#7]
We have the following snippet in our codebase
```
new XLSX()
        .extract(inFile, { sheet_name: mapping.sheetName })
        .on('row', row => writer.write(row))
        .on('error', error => {
            console.error(error);
            process.exit(1);
        })
        .on('end', () => {
            // Xlsx-extract might send end-event before its done reading rows. 
            // wait one second beofre assuming that prosessing is done.
            // If [Error: write after end] is observed during import, it's probably
            // writer.end() being run too soon
            setTimeout(() => {
                console.log('OK');
                writer.end();
            }, 1500);
        });
```

This is in order to minimize the amount of the following error:
```
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: write after end
    at writeAfterEnd (_stream_writable.js:159:12)
    at WriteStream.Writable.write (_stream_writable.js:204:5)
    at EsBulkImportWriter.writeln (/Users/tomasfagerbekk/Repos/HO21-datakilder/utilities/dist/EsBulkImportWriter.js:78:31)
    at EsBulkImportWriter.write (/Users/tomasfagerbekk/Repos/HO21-datakilder/utilities/dist/EsBulkImportWriter.js:71:22)
    at XLSX.<anonymous> (/Users/tomasfagerbekk/Repos/HO21-datakilder/utilities/dist/bin/preparebulk.js:36:27)
    at emitOne (events.js:77:13)
    at XLSX.emit (events.js:169:7)
```

It seems to us that 'row' events are being emitted before 'end' event in some cases (we do large bulk import). Do you know how we can fix this without the ugly workaround (which sometimes still break), or where we might have gone wrong?

**Comment by ffalt** [comment]
Hi, sorry for the delay. Must be a race condition between xml parsing & file closing. I'll have a look at it.

**Comment by ffalt** [comment]
@tomfa Hi, found the bug. For testing I slowed down the xml-parsing with a setTimeout and could reproduce your issue. It's fixed now in v0.0.0.12 & the current master branch.

**Comment by tomfa** [comment]
Great! thanks @ffalt
